xcodeproj/internal/xcodeprojinfo.bzl

```diff
@@ -256,6 +289,30 @@ def _skip_target(*, target, deps, deps_attrs, transitive_infos):
         ),
     )
 
+def _create_envs_depset(*, ctx, target, id, automatic_target_info):
+    raw_run_env = {}
+    if hasattr(native, "RunEnvironmentInfo"):
+        raw_run_env = target[RunEnvironmentInfo].environment if RunEnvironmentInfo in target else {}
+    test_env = getattr(ctx.rule.attr, automatic_target_info.env, {})
+
+    # Some keys are not applicable in schemes, we will filter them out here
+    _IGNORED_BAZEL_ENV_VARIABLES = [
+        "XCODE_VERSION_OVERRIDE",
+        "XCODE_VERSION",
+    ]
+    run_env = {
+        key: value
+        for key, value in raw_run_env.items()
+        if key not in _IGNORED_BAZEL_ENV_VARIABLES
+    }
+
+    return struct(
+        id = id,
+        env = struct(
+            **dicts.add(run_env, test_env, ctx.configuration.test_env)
```

Comment:
So `target[RunEnvironmentInfo].environment` didn't have `--test_env`? Does it have anything we want?

Reply:
I dont think we actually need it for anything 🤞